src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/custom.py

```diff
@@ -233,54 +229,50 @@ def acr_login(registry_name, resource_group_name=None, username=None, password=N
+    docker_not_installed = "Please verify if docker is installed."
+    docker_not_available = "Please verify if docker daemon is running properly."
+
     try:
-        call(["docker", "ps"], stdout=PIPE, stderr=PIPE)
-    except:
-        raise CLIError("Please verify whether docker is installed and running properly")
-
-    registry, _ = get_registry_by_name(registry_name, resource_group_name)
-    sku_tier = registry.sku.tier
-    login_server = registry.login_server
-
-    # 1. if username was specified, verify that password was also specified
-    if username and not password:
-        try:
-            password = prompt_pass(msg='Password: ')
-        except NoTTYException:
-            raise CLIError('Please specify both username and password in non-interactive mode.')
-
-    if sku_tier == SkuTier.managed.value:
-        # 2. if we don't yet have credentials, attempt to get a refresh token
-        if not password:
-            try:
-                username = "00000000-0000-0000-0000-000000000000"
-                password = get_login_refresh_token(login_server)
-            except Exception as e:  # pylint: disable=broad-except
-                logger.warning("AAD authentication failed with message: %s", str(e))
-
-    # 3. if we still don't have credentials, attempt to get the admin credentials (if enabled)
-    if not password:
-        try:
-            cred = acr_credential_show(registry_name)
-            username = cred.username
-            password = cred.passwords[0].value
-        except Exception as e:  # pylint: disable=broad-except
-            logger.warning("Admin user authentication failed with message: %s", str(e))
-
-    # 4. if we still don't have credentials, prompt the user
-    if not password:
-        try:
-            username = prompt('Username: ')
-            password = prompt_pass(msg='Password: ')
-        except NoTTYException:
-            raise CLIError(
-                'Unable to authenticate using AAD or admin login credentials. ' +
-                'Please specify both username and password in non-interactive mode.')
-
-    call(["docker", "login",
-          "--username", username,
-          "--password", password,
-          login_server])
+        p = Popen(["docker", "ps"], stdout=PIPE, stderr=PIPE)
+        returncode = p.wait()
+    except OSError:
+        raise CLIError(docker_not_installed)
+    except SubprocessError:
+        raise CLIError(docker_not_available)
+
+    if returncode:
+        raise CLIError(docker_not_available)
+
+    login_server, username, password = get_login_credentials(
+        registry_name=registry_name,
+        resource_group_name=resource_group_name,
+        username=username,
+        password=password)
+
+    use_password_stdin = False
+    try:
+        p = Popen(["docker", "login", "--help"], stdout=PIPE, stderr=PIPE)
+        stdout, _ = p.communicate()
+        if b'--password-stdin' in stdout:
+            use_password_stdin = True
```

Comment:
Is there other approach to determine if the underlying docker client support this parameter? For example, can you determine the version of the docker client? Parsing the help doc may not be the most stable or elegant approach. It also worth consider if there is localized docker client.

Reply:
It wasn't clear to us which version of `docker` has support for `--password-stdin`, besides `docker` has multiple editions. It is indeed not an elegant approach, but enumerating all versions/editions seem to be less stable than parsing the help message. I am not aware of localized docker client.

Reply:
ok.